server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -266,6 +267,26 @@ public static String threadName(final String nodeName, final String namePrefix)
         return "elasticsearch" + (nodeName.isEmpty() ? "" : "[") + nodeName + (nodeName.isEmpty() ? "" : "]") + "[" + namePrefix + "]";
     }
 
+    // to be used in assertions only.
+    public static boolean differentExecutors(Thread thread1, Thread thread2) {
```

Comment:
I had that first, but since it no longer asserts that the executors are different but rather returns whether they are, I renamed it to avoid confusion.